### DISCUSSION TITLE
Running on Raspberry Pi?

### DISCUSSION THREAD

**Discussion by Lavogel** [philosowaffle/peloton-to-garmin#152] [Q&A] [answered]
Hi!

First of all: thanks for this tool. I looked it up and got it running a few days ago on my PC via Docker and it works like a charm. 

Now I am wondering if there is a way (maybe a modified Docker image) to make it work on the arm64 architecture of a Raspberry Pi? Thank you for input!

**Comment by Vicore1** [comment, marked as answer]
I run in Docker on a Pi4 B without issue. Have you tried it?
uname -a
Linux 5.10.17-v8+ #1421 SMP PREEMPT Thu May 27 14:01:37 BST 2021 aarch64 GNU/Linux

**Reply by Lavogel** [reply]
When trying to pull the image with `docker pull philosowaffle/peloton-to-garmin:latest` I get the following response:
`no matching manifest for linux/arm/v7 in the manifest list entries`

I'm quiet unfamiliar with linux and a total docker newbie so maybe it is just the wrong approach.
uname says Linux raspberrypi 5.10.17-v7l+ #1414 SMP Fri Apr 30 13:20:47 BST 2021 armv7l GNU/Linux

**Reply by Vicore1** [reply]
It would appear (based on that uname) that you are using 32 bit Raspbian which is what the Raspberry Pi Imaginer will give you as a default.  If you want to use the existing docker imagines for this project you will need to switch out to the 64bit OS (which is in Beta), you can find here: https://downloads.raspberrypi.org/raspios_arm64/images/ you can use the Raspberry Pi Imaginer to write the imagine to a SD card.  Personally, I switched to the running on on one of my PIs when I decided to go to V2, I haven't looked back.

**Reply by Lavogel** [reply]
Doh! Raspberries are 64bit since model 3 IIRC and I never gave a thought about the fact that current OS builds for the model 4 could still be 32bit. I will give it a try and I am sure it will work. Thank you!

**Reply by Vicore1** [reply]
Reach back out if you run into any issues we can look at additional remediation. I assume at this time this can be marked answered when you get a chance.

**Reply by Lavogel** [reply]
I installed the 64 bit Raspian OS and Docker following the steps for Raspian at https://docs.docker.com/engine/install/debian/. I added a new user to the docker group and I am able to run the mentioned Hello World container with this user without sudo.
I also pulled the peloton-to-garmin-container without a problem now.

But when it comes to starting it up, I am running into the boundaries of my linux skills: 
1. The p2g-wiki mentions the two configuration files configuration.local.json and docker-compose.yaml but not where I have to place them (/home/username ?). 
2. When trying to run `docker compose up -d` I get "unknown shorthand flag: 'd' in -d". And when using `docker compose up` only it says: "docker: 'compose' is not a docker command.". I found a guide at https://docs.docker.com/compose/install/ to install composer and followed it too but still get the same result.

Maybe you have another advice how to solve this?

**Reply by Vicore1** [reply]
So locate where you run this from (/home/username will work, I personally dropped in it a sub-directory since I have a bunch of other containers running with a bunch of compose files and I am lazy).  Second, create your docker-compose.yaml (use the example [provided ](https://github.com/philosowaffle/peloton-to-garmin/blob/master/docker-compose.yaml)) in that location. Next, create your configuration.local.json  in that same directory with whatever configuration setting float your boat, [use the example](https://github.com/philosowaffle/peloton-to-garmin/blob/master/configuration.example.json). Finally, run there are two versions of docker compose commands depending in the version of compose you are using, I know that sounds dumb, it is very dumb. The old version uses  `docker-compose` and the new one uses `docker compose` (no hyphen), my guess is you are getting a "docker: 'compose' is not a docker command." you are using the inadvertently using the wrong command.  Try it as `docker-compose up -d` and see if that works.

**Reply by Vicore1** [reply]
Also just as an FYI for someone new to all this `-d` is to run the container in the background it is short to `--detach`

**Reply by Lavogel** [reply]
So it does not matter where I create the two config files as long as I execute the docker compose command from the same directory?
I tried the docker-compose command but got another error. When looking up the error I managed to identify the reason but I still got no solution for my main task.
In the docker compose installation article (https://docs.docker.com/compose/install/) the given command to download composer is `sudo curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-$(uname -s)-$(uname -m)" -o /usr/local/bin/docker-compose` which translates to  `sudo curl -L "https://github.com/docker/compose/releases/download/1.29.2/docker-compose-Linux-aarch64" -o /usr/local/bin/docker-compose` on my machine. Unfortunately there is no matching file nor compose branch for aarch64.

While searching the internet I'm stumbling about a variant of using pip and python and maybe more tools to make compose work under Raspi OS. Can you confirm this with your installation?

Sorry for bothering you with these things. The logic behind linux dependencies and all those differences between the distributions is something that does not seem to fit into my head.
And thanks again for your time so far!

**Reply by Vicore1** [reply]
I might try and remove it and reinstall  it trying to follow this [guide](https://dev.to/elalemanyo/how-to-install-docker-and-docker-compose-on-raspberry-pi-1mo)

**Reply by Lavogel** [reply]
That's it! Now I can get back to training. ;) Thank you!
docker container ls
CONTAINER ID   IMAGE                             COMMAND                  CREATED         STATUS         PORTS     NAMES
d076bed560e4   philosowaffle/peloton-to-garmin   "./PelotonToGarminCo…"   3 minutes ago   Up 3 minutes             p2g

**Reply by Vicore1** [reply]
Great!
